### PR TITLE
ACSP-233 - Add sample events to event registrations

### PR DESCRIPTION
**Pull request by jnatherley**
## Description

A sample template event is added to the event provider metadata as part of the onboarding script (`npm run onboard`) execution.

In the developer console, the sample event can be viewed and sent on the **Registration Details** tab in the **Subscribed Events** section

image

image

## Related Issue

N/A

## Motivation and Context

Adding the sample events functionality serves two main purposes:
* self-document the events managed by the system: a developer can get a quick reference of each event by clicking the *View sample payload* button
* test the system: a developer can test the integration by clicking the *Send sample event" button

## How Has This Been Tested?

The change has been tested by deploying a starter kit project on a clean App Builder project, running the onboard script and viewing and sending sample events.

## Screenshots (if appropriate):

Here's a screenshot showing the "Commerce Product Sync" sample events being send using the developer console UI and being successfully process by the system.
image

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- x ] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by rojoangel** [comment]
In f33734f made some changes to the sample event templates to
* fix some wrong attribute types
* make the sample data more realistic
* wrap events originating in commerce in a `value` object to make them acceptable by the system

**Review by rojoangel — APPROVED** [review]
LGTM with a couple of minor change requests

File: scripts/lib/metadata.js [hunk 1]
```diff
@@ -44,7 +56,25 @@ function buildProviderData (providerEvents) {
...
+
+  const sampleEvent = base64EncodedSampleEvent(sampleEventTemplate)
+  if (sampleEvent) {
+    body = {
+      // eslint-disable-next-line camelcase
+      ...body,
+      sample_event_template: sampleEvent
+    }
+  }
+
```
[the remark this answers is not in the document]
Comment (jnatherley) [reply scripts/lib/metadata.js]: Done in https://github.com/adobe/commerce-integration-starter-kit/pull/17/commits/442b61fc3604703e28d3d5c4c377a49194a5abf5

File: scripts/lib/metadata.js [hunk 2]
```diff
@@ -214,3 +238,4 @@ async function main (clientRegistrations, providers, environment, accessToken) {
 }
 
 exports.main = main
+exports.foo = {}
```
[the remark this answers is not in the document]
Comment (jnatherley) [reply scripts/lib/metadata.js]: Done in https://github.com/adobe/commerce-integration-starter-kit/pull/17/commits/b63c6ef2a1e16bb1941a1d515a52e82eec74ab25

**Pull request merged**